### PR TITLE
fix: reorder close file dialog to match convention

### LINKED ISSUE #2300: Improve saving dialog
__Is your feature request related to a problem? Please describe.__

When closing Modeler with a not saved model, the dialog displays the buttons in a not standardized way (compared to e.g. word) and the focus is on cancel. 
image

Due to this behavior, it happend to me several times, that I lost my model without saving.

__Describe the solution you'd like__

Implement default behavior like e.g. Word with the focus on save.
image

### PR DESCRIPTION
**Pull request by Skaiir**
Closes #2300

This requires local testing.

- [x] windows @Skaiir 
- [x] mac
- [x] linux @marstamm @nikku 

Artefacts

    https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/cfd-cons/camunda-modeler-cfd-cons-linux-x64.tar.gz
    https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/cfd-cons/camunda-modeler-cfd-cons-mac.dmg
    https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/cfd-cons/camunda-modeler-cfd-cons-mac.zip
    https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/cfd-cons/camunda-modeler-cfd-cons-win-ia32.zip
    https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/cfd-cons/camunda-modeler-cfd-cons-win-x64.zip

### CODE REVIEW

**Comment by nikku** [comment]
The main difference here is between Windows and MacOS (mac putting OK last, windows putting OK first), cf. [source](https://www.nngroup.com/articles/ok-cancel-or-cancel-ok/). 

On Linux this _may_ depend on the window manager / UI toolkit, but for me it is Mac-style (primary, forward looking action (i.e. OK) last) across numerous apps I've checked.

File: client/src/remote/Dialog.js [hunk 1]
```diff
@@ -113,12 +113,19 @@ export default class Dialog {
...
     } = options;
 
+    const buttons = [
+      { id: 'save', label: 'Save' },
+      { id: 'discard', label: 'Don\'t Save' },
+      { id: 'cancel', label: 'Cancel' },
+    ];
+
+    // Re-order buttons for linux
+    if (this.backend.getPlatform() == 'linux') {
```
Comment (nikku) [inline client/src/remote/Dialog.js:123]: Should be `!= windows` I believe, cf. https://github.com/camunda/camunda-modeler/pull/2895#issuecomment-1091429485.

**Comment by Skaiir** [comment]
@nikku Actually on modern mac with the nice overlays at least for this kind of menu it's windows style:

image
image

Although it doesn't matter here because Mac seems clever enough to do the ordering themselves, so I guess we should see how it looks on older mac versions?

Regarding Linux, we officially only support Ubuntu so we should align with that, right?

**Comment by christian-konrad** [comment]
@Skaiir what's the current state of this PR and what is blocking it from merge?

**Comment by Skaiir** [comment]
@christian-konrad Actually I just went on holiday and haven't followed up on this yet. I just need to verify this works as expected for most OSs and then we should be good to merge this :)

**Comment by marstamm** [comment]
I can confirm it works on ubuntu:

image

**Comment by Skaiir** [comment]
Unfortunately it does not with mac, my initially assumption was that mac re-ordered the buttons automatically, but that's only partially true. It only moves the cancel button to the end.

image

Hence we'll configure mac just like windows.

File: client/src/remote/Dialog.js (hunk 1 above)
Comment (Skaiir) [reply client/src/remote/Dialog.js:123]: We'll have to keep this as is, because newer mac versions are windows style, save first.

**Comment by pinussilvestrus** [comment]
Looks good on Mac 11.15

image

**Review by nikku — APPROVED** [review]

**Comment by nikku** [comment]
@Skaiir What is the status on this one? Do you want to wrap this up? From what I see all checks (except for codecov) succeed. Your call whether to fix or ignore the last.

**Comment by Skaiir** [comment]
@nikku Hey, sorry I thought this one would have gotten automatically merged. I'll rebase, and if nothing's wrong merge.

**Pull request merged**